proposals/spi-proposal.md

```diff
@@ -112,168 +105,150 @@ MSBFIRST = 1
 | 2       | 1                     | 0                  |
 | 3       | 1                     | 1                  |
 
-
 #### maxSpeed
 
-The maximum speed of communication with the SPI device. For a SPI device rated up to
-5 MHz, use 5000000.
-
-#### wordSize
+The maximum speed of communication with the SPI device. For a SPI device rated
+up to 5 MHz, use 5000000.
 
-The size of a `word` in bits. Typically 8-bits (default). 0 = DEFAULT = 8-bits, 1 = 1 bit,
-2 = 2 bits, etc (limit is TBD).
+*For platforms that use a clock divider instead of a speed, pass the clock
+divider value instead.* 
 
-#### csPin / csPinOptions
+#### wordSize
 
-Specify a CS (chip select) pin to automatically toggle the pin rather than having to
-send a separate `DIGITAL_MESSAGE` command with every transfer.
+The size of a `word` in bits. Typically 8-bits (default). 0 = DEFAULT = 8-bits,
+1 = 1 bit, 2 = 2 bits, etc (limit is TBD).
 
-The default behavior is to pull the CS pin LOW at the beginning of a transfer and toggle
-it at the end of the transfer. Setting bit 0 (`CS_ACTIVE_STATE`) will pull the CS pin HIGH
-at the start of the transfer. Set bit 1 (`CS_TOGGLE`) to handle the rare use case of
-toggling the CS pin between every word sent rather than at the end of the transaction.
+#### csPinOptions / csPin
 
-The CS pin behavior can be further controlled by using the `csPinControl` byte of the
-`SPI_TRANSFER`, `SPI_WRITE` or `SPI_READ` messages.
+Use `csPinOptions` to set the active state (typically LOW) for the CS pin. If
+the platform's SPI implementation does not already automatically handle the CS
+pin (it's handled automatically on Raspberry Pi, but not Arduino boards for
+example), then set `CS_PIN_CONTROL` to `enable` and specify the `csPin` number
+in byte 12. If the platform already handles the csPin then set
+`CS_PIN_CONTROL` to `disable` and the `csPin` number will be ignored (set to
+zero). For non-Linux platforms such as Arduino, to enable manual control of
+the CS pin using `DIGITAL_MESSAGE` commands, set `CS_PIN_CONTROL` to `disable`.
 
+### SPI_TRANSFER
 
-### SPI_END_TRANSACTION
+Full-duplex write/read transfer. This is the normal SPI transfer mode, a word
+must be written for every word read. Reply is sent via `SPI_REPLY`.
 
-Signals the end of a transaction. Must be sent before starting communication
-with another SPI device.
+Since transport (Serial, Ethernet, Wi-Fi, etc) buffers tend to be small on
```

Comment:
I may end up needing a `SPI_BEGIN_MESSAGE`, `SPI_END_MESSAGE` pair of commends to ensure proper framing of SPI messages for Arduino-based implementation. However I'm going to see if I can get away without it at first. Depends on the outcome of the Arduino implementation of this proposal.